src/root.component.js

```diff
@@ -31,8 +31,9 @@ export default function Root(props) {
   }, []);
 
   useEffect(() => {
+    const directLocalStorage = localStorage.getItem("single-spa-playground");
     if (
-      !localStorage.getItem("single-spa-playground") &&
+      !(directLocalStorage && directLocalStorage.application) &&
```

Comment:
`directLocalStorage` is a string, right? And strings won't have an `application` property on them. I think you need to JSON.parse() first?

Reply:
I will use the useLocalStorageData hook to get this data!